exchanges/exchange.go

```diff
@@ -47,6 +48,16 @@ var (
 	errSymbolCannotBeMatched             = errors.New("symbol cannot be matched")
 )
 
+// SetRequester sets the instance of the requester
+func (b *Base) SetRequester(r *request.Requester) error {
+	if r == nil {
+		return fmt.Errorf("%s cannot set requester, no requester provided", b.Name)
+	}
+
```

Comment:
Suggestion: We could add a check here to see if its already been set?

Reply:
I think there might be instances where it makes sense to replace this. For instance I like to just call the standard `SetupDefaults` that come with the exchange implementations which set a default Requester already, but want to replace it with my own.

Reply:
OK 👍